src/mono/sample/wasm/browser/main.js

```diff
@@ -22,6 +24,9 @@ async function main() {
             return {
                 disableDotnet6Compatibility: true,
                 configSrc: "./mono-config.json",
+                locateFile: (path, prefix) => {
+                    return window.location.origin + "/" + prefix + path;
```

Comment:
What does this help with, exactly?

Reply:
https://github.com/dotnet/runtime/blob/d93874fa9dc3a4e74f2adf555879ce4ec9911c14/src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs#L1276

Without this change the urls here are like this:
managed/MyAssembly.dll
managed/MyAssembly.pdb

and should be:
http://localhost:8080/managed/MyAssembly.dll
http://localhost:8080/managed/MyAssembly.pdb

Reply:
How does it work for other apps then?

Reply:
I really don't know I just printed the values while running debugger tests before and after calling localteFile:
https://github.com/dotnet/runtime/blob/c537cc6a45c4c75524ec98f6a8d4c9637438958e/src/mono/wasm/runtime/startup.ts#L489

And they are like this:
before: managed/MyAssembly.dll
after: http://localhost:8080/managed/MyAssembly.dll

Reply:
the production app could be running at any URL, not just `window.location.origin` 
Maybe some files could be even  on CDN on different host.

I guess the real questions is how the BrowserDebugProxy learned about this URL. We need to change the code in place which sent the URL to the BrowserDebugProxy. The change should try to make it absolute URL.

Reply:
This is how is done on Blazor: https://github.com/dotnet/aspnetcore/blob/8a95ee9b8b1bbd41357cd756375d78cec6936116/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L488

https://github.com/dotnet/aspnetcore/blob/8a95ee9b8b1bbd41357cd756375d78cec6936116/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L515

Reply:
At compile time of the sample, we generate the mono-config.json, and it's from this file that we push assemblies to the loaded_files property.
The format file is like this:
`{
  "assembly_root": "managed",
  "debug_level": 1,
  "assets": [
    {
      "behavior": "assembly",
      "name": "Microsoft.CSharp.dll"
    },
    {
      "behavior": "assembly",
      "name": "Microsoft.CSharp.pdb"
    },
    {
      "behavior": "assembly",
      "name": "Microsoft.VisualBasic.Core.dll"
    },
    {
      "behavior": "assembly",
      "name": "Microsoft.VisualBasic.Core.pdb"
    }`

And at compile time we don't know the absolute URL to make it correct.

Reply:
I think we need to have something like Blazor's `toAbsoluteUrl` on runtime at place which sends the url to proxy.
Or maybe the we need to unify the url during the download.